### PR TITLE
feat(medusa-js, medusa-react): PublishableKeys SC management

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- add the support for managing sales channel <> publishable key scopes to `medusa-js` and `medusa-react`

---

RESOLVES CORE-817

### CODE REVIEW

File: packages/medusa-react/src/hooks/admin/publishable-api-keys/queries.ts [hunk 1]
```diff
@@ -52,3 +54,21 @@ export const useAdminPublishableApiKeys = (
...
+  query?: GetPublishableApiKeySalesChannelsParams,
+  options?: UseQueryOptionsWrapper<
+    Response<AdminSalesChannelsListRes>,
+    Error,
+    ReturnType<PublishableApiKeyQueryKeys["detail"]>
+  >
+) => {
+  const { client } = useMedusa()
+  const { data, ...rest } = useQuery(
+    adminPublishableApiKeysKeys.detail(id),
```
Comment (pKorsholm) [inline packages/medusa-react/src/hooks/admin/publishable-api-keys/queries.ts:69]: We might run in to some conflicts with the caching key used here if the publishablekey is fetched using the `useAdminPublishableApiKey` hook before trying to fetch the sales channels at this point. 

To solve it I think we should use a different key for this as we also do with listing products by price list (as well as some other places). wdyt?

File: packages/medusa-react/src/hooks/admin/publishable-api-keys/mutations.ts [hunk 2]
```diff
@@ -97,3 +98,53 @@ export const useAdminRevokePublishableApiKey = (
...
+  const queryClient = useQueryClient()
+
+  return useMutation(
+    (payload: AdminPostPublishableApiKeySalesChannelsBatchReq) =>
+      client.admin.publishableApiKeys.deleteSalesChannelsBatch(id, payload),
+    buildOptions(
+      queryClient,
+      [
+        adminPublishableApiKeysKeys.lists(),
+        adminPublishableApiKeysKeys.detail(id),
```
Comment (pKorsholm) [inline packages/medusa-react/src/hooks/admin/publishable-api-keys/mutations.ts:145]: adjust keys depending on the outcome of my other comment

File: packages/medusa-react/src/hooks/admin/publishable-api-keys/mutations.ts [hunk 3]
```diff
@@ -97,3 +98,53 @@ export const useAdminRevokePublishableApiKey = (
...
+  const queryClient = useQueryClient()
+
+  return useMutation(
+    (payload: AdminPostPublishableApiKeySalesChannelsBatchReq) =>
+      client.admin.publishableApiKeys.addSalesChannelsBatch(id, payload),
+    buildOptions(
+      queryClient,
+      [
+        adminPublishableApiKeysKeys.lists(),
+        adminPublishableApiKeysKeys.detail(id),
```
Comment (pKorsholm) [inline packages/medusa-react/src/hooks/admin/publishable-api-keys/mutations.ts:120]: adjust keys depending on the outcome of my other comment

**Review by pKorsholm — COMMENTED** [review]
One comment regarding the caching keys but otherwise it looks good! 😄

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**